.travis.yml

```diff
@@ -12,42 +12,82 @@ addons:
     - virt-what
     update: true
 
+git:
+  submodules: false
+  quiet: true
+
 services:
   - docker
 
+stages:
+  - Dev & Unit tests
+  - BATS tests
+  - Run unattended openHABian installation
+  - Shellcheck
+
 install:
-  - shellcheck --version
-# some tests such as zram.bats may fail if we run in emulation mode (testing for ARM emulated on x86 Travis cloud servers)
-  - sudo apt-get install virt-what && VIRT=$(sudo virt-what) && echo "Travis VM running on ${VIRT:-native HW}."
-  - docker run --rm --privileged multiarch/qemu-user-static:register --reset
 # prepare configuration for tests, select debug level here:
   - sed -iE 's#mode=.*$#mode=unattended_debug#' build-image/openhabian.conf
   - #sed -iE 's#mode=.*$#mode=debug_maximum#' build-image/openhabian.conf
+  - echo "Cloning openHABian repo from https://github.com/${TRAVIS_REPO_SLUG}.git"
   - sed -i 's#repositoryurl=https://github.com/openhab/openhabian.git#repositoryurl=https://github.com/'${TRAVIS_REPO_SLUG}'.git#' build-image/openhabian.conf
   - sed -i 's#clonebranch=master#clonebranch='${TRAVIS_BRANCH}'#' build-image/openhabian.conf
-  - docker build -t openhabian/rpi-openhabian -f Dockerfile.openhabian .
 
 jobs:
   fast_finish: true
   include:
-    - stage: Dev+unit-tests
-      script: docker run -it openhabian/rpi-openhabian bash -c 'bats -r -f "dev-." .'
-    - script: docker run -it openhabian/rpi-openhabian bash -c 'sudo bats -r -f "unit-." .'
-    - stage: Openhabian-Setup
+    - stage: Dev & Unit tests
+      script:
+        - sed -i 's/java_arch=32-bit$/java_arch=64-bit/' build-image/openhabian.conf
+        - docker build -t openhabian/generic-openhabian -f Dockerfile.openhabian-x86 .
+        - docker run -it openhabian/generic-openhabian bash -c '/usr/local/bin/bats -r -f "dev-." .'
+        - docker run -it openhabian/generic-openhabian bash -c '/usr/local/bin/bats -r -f "unit-." .'
+    - stage: Run unattended openHABian installation
+      env: amd64
+      script:
+        - export HW=x86
+        - sed -i 's/java_arch=32-bit$/java_arch=64-bit/' build-image/openhabian.conf
+        - docker build -t openhabian/${HW}-openhabian -f Dockerfile.openhabian-x86 .
+        - docker run --name "openhabian-${HW}" --privileged -d openhabian/${HW}-openhabian;
+          docker exec -it "openhabian-${HW}" bash -c \
```

Comment:
Maybe worth looking at the option allow_failures instead of hiding the fail here.
What I see at https://github.com/openhab/openhabian/pull/684/checks?check_run_id=610534466 is quite misleading....
Or do you think it is mandatory to work on all targets in the future?

Reply:
Well a little difficult to trade-off today. We need to discuss and eventually might change this over time, but for the start, IMO _testing_ aarch64 is required but _passing_ is optional as neither emulation nor (alpha) native compilation seem to properly work yet and I don't know if that is due to openhabian, Travis or even something else.
But we shouldn't remove this test to not lose the comprehensive view on test status.
On the remaining tests, I'd say passing is mandatory since we want to support all of those hardwares.

What do you think is misleading there ?
Granted the "|| true" in aarch64 is somewhat hidden, I will try allow_failures instead.
But it's always a lengthy trial-and-error game to make structurtal changes to the Travis config, so I don't want to spend too much time on "why don't you ..." trials.